### PR TITLE
Connect MemberApiV2 from Peatio

### PR DESCRIPTION
**Pull request by ymasiuk**

### CODE REVIEW

File: app/api/api_v1/withdraw.rb [hunk 1]
```diff
@@ -18,17 +18,12 @@ class Withdraw < Grape::API
...
     end
     post '/withdraws' do
-      if params[:currency_type] == 'fiat'
-        count = Beneficiary.active
-                           .where(uid: env['api.v1.authenticated_uid'],
-                                  rid: params[:rid]).count
-        error!('Beneficiary is not found', 404) if count < 1
+      currency = Peatio::MemberApiV2::Client.new.get_currency(params[:currency])
+      if currency['type'] == 'fiat' &&
+         !Beneficiary.active.where(uid: env['api.v1.authenticated_uid'], rid: params[:rid]).exists?
```
Comment (ysv) [inline app/api/api_v1/withdraw.rb]: Return 422 if beneficiary is not found

File: app/services/peatio/member_api_v2/client.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+module Peatio
+  module MemberApiV2
```
Comment (ysv) [inline app/services/peatio/member_api_v2/client.rb]: `MemberAPIv2` and add acronym https://github.com/rubykube/applogic/blob/master/config/initializers/inflections.rb#L19

File: app/services/peatio/member_api_v2/client.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      def request(request_method, request_path)
+        unless request_method.in?(%i[get])
+          raise ArgumentError, "Request method is not supported: #{request_method.inspect}."
+        end
+
+        http_client
+            .public_send(request_method, build_path(request_path))
+            .tap { |response| response unless response.success? }
+            .assert_success!
+            .body
```
Comment (ysv) [inline app/services/peatio/member_api_v2/client.rb]: rescue errors (same as ManagementAPIv2)

File: app/services/peatio/member_api_v2/client.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+module Peatio
+  module MemberApiV2
+    class Client
```
Comment (ysv) [inline app/services/peatio/member_api_v2/client.rb]: public methods should be initialize and get_currency other private

File: app/services/peatio/member_api_v2/client.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      def build_path(path)
+        "api/v2/#{path}"
+      end
+
+      def get_currency(currency)
+        request(:get, "currencies/#{currency}")
+      end
+    end
+  end
+end
```
Comment (ysv) [inline app/services/peatio/member_api_v2/client.rb]: Add blank line

File: spec/api/v1/membem_api_v2_spec.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+      it 'send request with fiat' do
+        expect(Peatio::MemberApiV2::Client.new.get_currency(currency['id'])).to eq currency
+      end
+    end
+
+    context 'cryptocurrency' do
+      let(:currency) do
+        {
+            'id' => 'btc', 'symbol' => '$', 'type' => 'coin', 'deposit_fee' => '0.0',
```
Comment (ysv) [inline spec/api/v1/membem_api_v2_spec.rb]: Identify with 2 spaces (not 4)

File: spec/api/v1/withdraw_spec.rb [hunk 7]
```diff
@@ -27,13 +27,27 @@
         .to_return(status: barong_response.status,
                    body: barong_response.body.to_json.to_s,
                    headers: {})
+      stub_request(:get, "#{ENV.fetch('PEATIO_ROOT_URL')}/api/v2/currencies/BTC")
```
Comment (ysv) [inline spec/api/v1/withdraw_spec.rb]: Use downcase currencies codes

**Comment by ymasiuk** [comment]
@ysv review please

**Comment by ysv** [comment]
reopened in #41

**Pull request closed without merging**